shared/Microsoft.Extensions.TypeNameHelper.Sources/TypeNameHelper.cs

```diff
@@ -37,121 +35,88 @@ public static string GetTypeDisplayName(object item, bool fullName = true)
 
         public static string GetTypeDisplayName(Type type, bool fullName = true)
         {
-            var sb = new StringBuilder();
-            ProcessTypeName(type, sb, fullName);
-            return sb.ToString();
+            var builder = new StringBuilder();
+            ProcessTypeName(builder, type, fullName);
+            return builder.ToString();
         }
 
-        private static void AppendGenericArguments(Type[] args, int startIndex, int numberOfArgsToAppend, StringBuilder sb, bool fullName)
+        private static void ProcessTypeName(StringBuilder builder, Type type, bool fullName)
         {
-            var totalArgs = args.Length;
-            if (totalArgs >= startIndex + numberOfArgsToAppend)
+            if (type.IsGenericType)
             {
-                sb.Append("<");
-                for (int i = startIndex; i < startIndex + numberOfArgsToAppend; i++)
-                {
-                    ProcessTypeName(args[i], sb, fullName);
-                    if (i + 1 < startIndex + numberOfArgsToAppend)
-                    {
-                        sb.Append(", ");
-                    }
-                }
-                sb.Append(">");
+                var genericArguments = type.GetGenericArguments();
+                ProcessGenericType(builder, type, genericArguments, genericArguments.Length, fullName);
             }
-        }
-
-        private static void ProcessTypeName(Type t, StringBuilder sb, bool fullName)
-        {
-            var typeInfo = t.GetTypeInfo();
-
-            if (typeInfo.IsGenericType)
-            {
-                ProcessNestedGenericTypes(t, sb, fullName);
-            }
-            else if (typeInfo.IsArray)
+            else if (type.IsArray)
             {
-                ProcessArrayTypes(t, sb, fullName);
+                ProcessArrayType(builder, type, fullName);
             }
-            else if (_builtInTypeNames.ContainsKey(t))
+            else if (_builtInTypeNames.TryGetValue(type, out var buildInName))
             {
-                sb.Append(_builtInTypeNames[t]);
-            }
-            else
-            {
-                sb.Append(fullName ? t.FullName : t.Name);
+                builder.Append(buildInName);
```

Comment:
I think the else part here got removed causing test failures.

Reply:
Yes, thank you, currently in a process of fixing.